tests/system.rs

```diff
@@ -66,6 +67,27 @@ fn idempotence_tests() {
     assert!(fails == 0, "{} idempotent tests failed", fails);
 }
 
+// Run rustfmt in tidy mode and check we don't modify the file.
+#[test]
+fn tidy_tests() {
```

Comment:
I know it's a test, but there's a lot of code duplication between this and `idempotence_tests`.